src/libsystemd/sd-bus/bus-container.c

```diff
@@ -75,8 +75,6 @@ int bus_container_connect_socket(sd_bus *b) {
         r = wait_for_terminate_and_check("(sd-buscntrns)", child, 0);
         if (r < 0)
                 return r;
-        if (r != EXIT_SUCCESS)
```

Comment:
hmm, so previously it happened way too often that people inserted other uses of `r` (since that's auch catchall error variable) between setting and using the variable like you do here, if the setting and the user is so long apart.

hence, please add a temporay bool here:

```c
bool nonzero_exit_status = r != EXIT_SUCCESS;
```

And then when you want to check things below, do a:

```c
if (nonzero_exit_status)
        return -EPROTO;
```

that way if people start to use `r` for somethingelse in between it doesn't matter.